code/modules/mob/living/silicon/silicon_defense.dm

```diff
@@ -140,4 +137,7 @@
 
 /mob/living/silicon/flash_act(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0, type = /atom/movable/screen/fullscreen/flash/static)
 	if(affect_silicon)
-		return ..()
+		overlay_fullscreen("flash", type)
+		addtimer(CALLBACK(src, .proc/clear_fullscreen, "flash", 25), 70) //Flashes now blind silicons for the duration they used to stun
```

Comment:
SECONDS defines

Reply:
I hate that this just assumes the flash should be used for this length of time, is there no way to get the stun/flash duration from the caller or flash that was used?

Reply:
There is, but I'd need to pass it as an arg which is arguably even more messy. I'll do it if you insist, though.